pkg/controllermanager/controller/controllerregistration/controllerregistration_control.go

```diff
@@ -23,32 +23,31 @@ import (
 
 	"github.com/sirupsen/logrus"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/client-go/tools/cache"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-func (c *Controller) controllerRegistrationAdd(obj interface{}) {
+func (c *Controller) controllerRegistrationAdd(ctx context.Context, obj interface{}) {
 	key, err := cache.MetaNamespaceKeyFunc(obj)
 	if err != nil {
 		return
 	}
 	c.controllerRegistrationQueue.Add(key)
 
-	seedList, err := c.seedLister.List(labels.Everything())
-	if err != nil {
+	seedList := &gardencorev1beta1.SeedList{}
```

Comment:
Should we use a metadata-only list here?

Reply:
For sure, nice catch. Done with 14d2747